sstable/reader.go

```diff
@@ -554,13 +558,16 @@ var propertiesBlockBufPools = sync.Pool{
 	},
 }
 
-// ReadPropertiesBlock reads the properties block from the table.
-// We always read the properties block into a buffer pool instead
-// of the block cache.
-func (r *Reader) ReadPropertiesBlock(
-	ctx context.Context, bufferPool *block.BufferPool, deniedUserProperties map[string]struct{},
+// ReadPropertiesBlockExcludeUserProps reads the properties block
+// from the table. We always read the properties block into a buffer pool
+// instead of the block cache. Note that UserProperties should not be read
+// from returned properties as they do not filter out denied user properties
+// (those that pebble doesn't care about). UserProperties should be read
+// directly from the Reader struct.
+func (r *Reader) ReadPropertiesBlockExcludeUserProps(
```

Comment:
It was a little cumbersome to have callers pass this map in when most of them don't read `UserProperties`. I opted to change this function to exclude it for the common case but I might have to introduce it again later (maybe as a separate fn?) as there at least one place ([getTableProperties in filecache](https://github.com/cockroachdb/pebble/blob/master/file_cache.go#L864)) where we want a complete correctly populated `Properties`.

I could also keep the param and export `noDeniedUserProperties`. LMK what y'all think.

Reply:
Followed up with @jbowens offline and we can probably just get rid of `DeniedUserProperties` entirely and just use `ignoredInternalProperties` directly. I saw that in the PR introducing this the goal seemed to be reducing the mem footprint of the properties struct anyways -- https://github.com/cockroachdb/pebble/commit/2f73929887f0b57336ad29bed0e648f68579ad0d.